### DISCUSSION TITLE
Add option to enable foreign key links even when table schema lacks definition (for reasons)

### DISCUSSION THREAD

**Discussion by rngtng** [Sequel-Ace/Sequel-Ace#1998] [Ideas]
I love the option to follow a foreign key link by clicking on the arrow. But if I'm not wrong, this unfortunately requires the table schema to have the FK explicitly defined. 
Now I'm dealing with lots of (ruby on rails) project where this explicit definition is missing, hence Sequel-Ace not offering the link. Even though this is a goods practise, I can't enforce on the projects to add the missing definitions.

Now the idea/question is if an option for 'manual extension/overwrite' could be introduced to hint sequel ace the FK relations allowing to display the links. Instead of saving the relation to DB, it would save the relations to the connection file, similar as saving favourite queries..

wyt? Thx a lot!

**Comment by rngtng** [comment]
btw. I'd be happy to contribute a POC on this, unfortunately don't have any objC background and kinda stop to understand beyond https://github.com/Sequel-Ace/Sequel-Ace/blob/main/Source/Controllers/MainViewControllers/TableRelations/SPTableRelations.m :)

**Comment by Jason-Morcos** [comment]
Definitely open to review a PR but could be pretty complicated to implement - these relations would need to be cached somewhere locally, maybe in SQLite like we're doing with the query history

**Comment by rngtng** [comment]
Just learned, in DataGrip they call this virtual-foreign-keys, which would sufficient (for me) to cover my described usecase.. https://www.jetbrains.com/help/idea/virtual-foreign-keys.html
